ArcGISRuntimeSDKQt_CppSamples/EditData/EditFeaturesWithFeatureLinkedAnnotation/EditFeaturesWithFeatureLinkedAnnotation.cpp

```diff
@@ -0,0 +1,272 @@
+
+void EditFeaturesWithFeatureLinkedAnnotation::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<EditFeaturesWithFeatureLinkedAnnotation>("Esri.Samples", 1, 0, "EditFeaturesWithFeatureLinkedAnnotationSample");
+}
+
+MapQuickView* EditFeaturesWithFeatureLinkedAnnotation::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void EditFeaturesWithFeatureLinkedAnnotation::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  connect(m_mapView, &MapQuickView::mouseClicked, this, &EditFeaturesWithFeatureLinkedAnnotation::onMouseClicked);
+  connect(m_mapView, &MapQuickView::identifyLayersCompleted, this, &EditFeaturesWithFeatureLinkedAnnotation::onIdentifyLayersCompleted);
+
+  emit mapViewChanged();
+}
+
+void EditFeaturesWithFeatureLinkedAnnotation::onGeodatabaseDoneLoading(Error error)
+{
+  if (!error.isEmpty())
+    return;
+
+  GeodatabaseFeatureTable* pointFeatureTable = m_geodatabase->geodatabaseFeatureTable("Loudoun_Address_Points_1");
+  GeodatabaseFeatureTable* parcelLinesFeatureTable = m_geodatabase->geodatabaseFeatureTable("ParcelLines_1");
+
+  GeodatabaseFeatureTable* addressPointsAnnotationTable = m_geodatabase->geodatabaseAnnotationTable("Loudoun_Address_PointsAnno_1");
+  GeodatabaseFeatureTable* parcelLinesAnnotationTable = m_geodatabase->geodatabaseAnnotationTable("ParcelLinesAnno_1");
+
+  // create feature layers from tables in the geodatabase
+  m_pointFeatureLayer = new FeatureLayer(pointFeatureTable, this);
+  m_parcelLinesFeatureLayer = new FeatureLayer(parcelLinesFeatureTable, this);
+
+  // create annotation layers from tables in the geodatabase
+  m_addressPointsAnnotationLayer = new AnnotationLayer(addressPointsAnnotationTable, this);
+  m_parcelLinesAnnotationLayer = new AnnotationLayer(parcelLinesAnnotationTable, this);
+
+  // add feature layers to map
+  m_map->operationalLayers()->append(m_pointFeatureLayer);
+  m_map->operationalLayers()->append(m_parcelLinesFeatureLayer);
+
+  // add annotation layers to map
+  m_map->operationalLayers()->append(m_addressPointsAnnotationLayer);
+  m_map->operationalLayers()->append(m_parcelLinesAnnotationLayer);
+}
+
+void EditFeaturesWithFeatureLinkedAnnotation::onMouseClicked(QMouseEvent mouseEvent)
+{
+  clearSelection();
+
+  if (m_selectedFeature)
+  {
+    // move feature to clicked locaiton if already selected
+    const Point clickedPoint = m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+    moveFeature(clickedPoint);
+  }
+  else
+  {
+    // identify and select feature
+    m_mapView->identifyLayers(mouseEvent.x(), mouseEvent.y(), 10, false);
+  }
+}
+
+void EditFeaturesWithFeatureLinkedAnnotation::onIdentifyLayersCompleted(QUuid, QList<IdentifyLayerResult *> identifyResults)
```

Comment:
I believe this will be a space leak - every time this slot is called, the memory won't be deallocated. My understanding is that QList does not take ownership of its contents. We have similar situations in other samples, such as https://github.com/Esri/arcgis-runtime-samples-qt/blob/13981feb89980a56d00d1080d53242e4bae37766/ArcGISRuntimeSDKQt_CppSamples/EditData/UpdateGeometryFeatureService/UpdateGeometryFeatureService.cpp#L153